src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -200,6 +220,18 @@ def apply_mixins(cls, data: Dict, info: ValidationInfo) -> Dict:
                     mixin_defs=data["mixins"],
                 )
                 entities[entity_name] = merged_values
+                if entity["type"] == "application package" and "children" in entity:
```

Comment:
we don't really need the type check here, do we? If we simply omit it, the feature becomes a general one at essentially no cost.